### PR TITLE
- Added 'canonical transparency' cvar r_canontrans - this simply turn…

### PR DESCRIPTION
**Pull request by raa-eruanna**
…s off transparency for Doom objects that were marked as transparent sometime in ZDoom's development cycle

### CODE REVIEW

File: src/actor.h [hunk 1]
```diff
@@ -402,6 +402,7 @@ enum ActorFlag8
 {
 	MF8_FRIGHTENING		= 0x00000001,	// for those moments when halloween just won't do
 	MF8_INSCROLLSEC		= 0x00000002,	// actor is partially inside a scrolling sector
+	MF8_ZDOOMTRANS		= 0x00000004,	// is not normally transparent in Vanilla Doom
```
Comment (Gaerzi) [inline src/actor.h]: Shouldn't it be a render flag instead? (Same for SPRITEFLIP, actually.)

**Comment by coelckers** [comment]
What about Heretic, Hexen and Stife?

**Comment by raa-eruanna** [comment]
Oh, initially I thought you said they weren't needed...

Well - for those, I can do a +ZDOOMADD flag that flips them back to transparent. That's certainly doable. But you did not address Gez's concern - and I am not sure if I should, either. Admittedly, I was slightly unhappy to find out that it even existed, but it makes sense, I suppose...

**Comment by coelckers** [comment]
Sorry, but this is wrong. All additive translucency in Heretic and Hexen is for actors that were originally opaque. In not a single case there was a change from translucent to additive.

**Comment by raa-eruanna** [comment]
Same for Strife?

**Comment by coelckers** [comment]
Not sure about everything but for the vast majority, yes.

**Comment by raa-eruanna** [comment]
If there's a need to recover +ZDOOMADD it's in the commit history.

**Pull request merged**